test/Module/CreateTest.php

```diff
@@ -87,10 +101,24 @@ public function testErrorsWhenCannotCreateModuleTemplatesDirectory(): void
     {
         $baseModulePath = sprintf('%s/my-modules/MyApp', $this->dir->url());
 
+        $expectedArguments = [
+            $baseModulePath,
+            $baseModulePath . '/src',
+            $baseModulePath . '/templates',
+        ];
+
         $mkdir = $this->getFunctionMock('Mezzio\Tooling\Module', 'mkdir');
         $mkdir
             ->expects(self::exactly(3))
-            ->withConsecutive([$baseModulePath], [$baseModulePath . '/src'], [$baseModulePath . '/templates'])
+            ->with(self::callback(static function (mixed $arg) use ($expectedArguments): bool {
+                self::assertTrue(in_array(
+                    $arg,
+                    $expectedArguments,
+                    true,
+                ));
+
+                return true;
+            }))
```

Comment:
This updated test does not check that every argument was used. I would recommend to pass expected arguments by reference and unset from array when invoked

assert inside callback could potentially be swallowed by try-catch in tested code. I handled that somewhere, I'll find a link shortly.

Reply:
Here is an example that is safe from try-catch in tested code: https://github.com/laminas/laminas-skeleton-installer/blob/df6f117342589a94f54fb0851fefd767117fb668/test/OptionalPackagesInstallerTest.php#L119-L173

`$constraintFailures` array is used to catch and re-throw assertion failures from test context.
Phpunit mocking does that behind the scenes but only with expectations.

Reply:
We can't use `$matcher->numberOfInvokations()` here because we're still on PHPUnit 9.x

I've adjusted the test to assert argument order - I don't fancy upgrading this to PHPUnit 10 right now.